### DISCUSSION TITLE
Is it possible to use Retrieval with Azure OpenAI Service?

### DISCUSSION THREAD

**Discussion by k-ishii1020** [danny-avila/LibreChat#2533] [Q&A] [answered]
Does anyone know if Retrieval can be used with Azure OpenAI Service?

When I experimented with gpt-4-1106-preview in the Australia East region, the following error returned.

```
LibreChat | 2024-04-25 10:53:57 error: [/assistants] Error creating assistant 400 Retrieval tools are not supported.
```

**Comment by danny-avila** [comment, marked as answer]
> Does anyone know if Retrieval can be used with Azure OpenAI Service?
> 
> When I experimented with gpt-4-1106-preview in the Australia East region, the following error returned.
> 
> ```
> LibreChat | 2024-04-25 10:53:57 error: [/assistants] Error creating assistant 400 Retrieval tools are not supported.
> ```

No, Azure doesn't support the retrieval tool. They are maybe holding off until they adopt V2 which reworks retrieval

https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/assistant

**Reply by k-ishii1020** [reply]
I understand. Thank you for teaching me
